apis/python/src/tiledbsoma/_dataframe.py

```diff
@@ -172,16 +172,22 @@ def create(
                 index column name is required.
             domain:
                 An optional sequence of tuples specifying the domain of each
-                index column. Each tuple should be a pair consisting of the minimum and
-                maximum values storable in the index column. For example, if there is a
-                single int64-valued index column, then ``domain`` might be ``[(100,
-                200)]`` to indicate that values between 100 and 200, inclusive, can be
-                stored in that column.  If provided, this sequence must have the same
-                length as ``index_column_names``, and the index-column domain will be as
-                specified.  If omitted entirely, or if ``None`` in a given dimension,
-                the corresponding index-column domain will use the minimum and maximum
-                possible values for the column's datatype.  This makes a
-                :class:`DataFrame` growable.
+                index column. Each tuple must be a pair consisting of the
+                minimum and maximum values storable in the index column. For
+                example, if there is a single int64-valued index column, then
+                ``domain`` might be ``[(100, 200)]`` to indicate that values
+                between 100 and 200, inclusive, can be stored in that column.
+                If provided, this sequence must have the same length as
+                ``index_column_names``, and the index-column domain will be as
+                specified.  If omitted entirely, or if ``None`` in a given
+                dimension, the corresponding index-column domain will use the
+                smallest possible values for the column's datatype, and data
+                writes after that will fail with "A range was set outside of the
+                current domain". Unless you have a particular reason not to, you
+                should always provide the desired `domain` at create time: this
+                is an optional but strongly recommended parameter. See also
+                ``change_domain`` which allows you to expand the domain after
+                create.
```

Comment:
When I ran into this recently, omitting the domain resulted in `(0, -1)`, and subsequent writes failed (as noted here).

1. Should "the smallest possible values" instead say "an empty range", or similar?
2. Can we mention here how to subsequently grow the domain, when it's created as an empty set?

Reply:
@ryan-williams 
1. I'll change this, thanks
2. That's the sentence about `change_domain` is trying to do. Can you help me make it more clear to you?

Reply:
@ryan-williams commit for #1: https://github.com/single-cell-data/TileDB-SOMA/pull/3396/commits/7f1f774ceaef83f777665d18758057d7d0fe66eb